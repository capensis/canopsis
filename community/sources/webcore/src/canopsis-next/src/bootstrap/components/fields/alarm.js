export { default as CAlarmInfosAttributeField } from '@/components/forms/fields/alarm/c-alarm-infos-attribute-field.vue';
export { default as CAlarmTagField } from '@/components/forms/fields/alarm/c-alarm-tag-field.vue';
export { default as CAlarmField } from '@/components/forms/fields/alarm/c-alarm-field.vue';
