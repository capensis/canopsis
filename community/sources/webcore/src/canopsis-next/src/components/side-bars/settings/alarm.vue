<template lang="pug">
  div
    v-list.pt-0(expand)
      field-title(v-model="settings.widget.title", :title="$t('common.title')")
      v-divider
      field-periodic-refresh(v-model="settings.widget.parameters.periodic_refresh")
      v-divider
      v-list-group(data-test="advancedSettings")
        v-list-tile(slot="activator") {{ $t('settings.advancedSettings') }}
        v-list.grey.lighten-4.px-2.py-0(expand)
          field-default-sort-column(
            v-model="settings.widget.parameters.sort",
            :columns="settings.widget.parameters.widgetColumns",
            :columns-label="$t('settings.columnName')"
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.widgetColumns",
            :label="$t('settings.columnNames')",
            with-html,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.widgetGroupColumns",
            :label="$t('settings.groupColumnNames')",
            with-html,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.serviceDependenciesColumns",
            :label="$t('settings.trackColumnNames')",
            with-color-indicator
          )
          v-divider
          field-default-elements-per-page(v-model="settings.userPreferenceContent.itemsPerPage")
          v-divider
          field-opened-resolved-filter(v-model="settings.widget.parameters.opened")
          v-divider
          template(v-if="hasAccessToListFilters")
            field-filters(
              v-model="settings.widget.parameters.mainFilter",
              :entities-type="$constants.ENTITIES_TYPES.alarm",
              :filters.sync="settings.widget.parameters.viewFilters",
              :condition.sync="settings.widget.parameters.mainFilterCondition",
              :has-access-to-add-filter="hasAccessToAddFilter",
              :has-access-to-edit-filter="hasAccessToEditFilter",
              @input="updateMainFilterUpdatedAt"
            )
            v-divider
          template(v-if="hasAccessToListRemediationInstructionsFilters")
            field-remediation-instructions-filters(
              v-model="settings.widget.parameters.remediationInstructionsFilters",
              :has-access-to-add-filter="hasAccessToAddRemediationInstructionsFilter",
              :has-access-to-edit-filter="hasAccessToEditRemediationInstructionsFilter"
            )
            v-divider
          field-live-reporting(v-model="settings.widget.parameters.liveReporting")
          v-divider
          field-info-popup(
            v-model="settings.widget.parameters.infoPopups",
            :columns="settings.widget.parameters.widgetColumns"
          )
          v-divider
          field-text-editor(
            data-test="widgetMoreInfoTemplate",
            v-model="settings.widget.parameters.moreInfoTemplate",
            :title="$t('settings.moreInfosModal')"
          )
          v-divider
          field-grid-range-size(
            v-model="settings.widget.parameters.expandGridRangeSize",
            :title="$t('settings.expandGridRangeSize')"
          )
          v-divider
          field-switcher(
            data-test="isHtmlEnabledOnTimeLine",
            v-model="settings.widget.parameters.isHtmlEnabledOnTimeLine",
            :title="$t('settings.isHtmlEnabledOnTimeLine')"
          )
          v-divider
          v-list-group(data-test="ackGroup")
            v-list-tile(slot="activator") Ack
            v-list.grey.lighten-4.px-2.py-0(expand)
            field-switcher(
              data-test="isAckNoteRequired",
              v-model="settings.widget.parameters.isAckNoteRequired",
              :title="$t('settings.isAckNoteRequired')"
            )
            v-divider
            field-switcher(
              data-test="isMultiAckEnabled",
              v-model="settings.widget.parameters.isMultiAckEnabled",
              :title="$t('settings.isMultiAckEnabled')"
            )
            v-divider
            field-fast-ack-output(v-model="settings.widget.parameters.fastAckOutput")
          v-divider
          field-switcher(
            v-model="settings.widget.parameters.isSnoozeNoteRequired",
            :title="$t('settings.isSnoozeNoteRequired')"
          )
          v-divider
          field-enabled-limit(
            v-model="settings.widget.parameters.linksCategoriesAsList",
            :title="$t('settings.linksCategoriesAsList')",
            :label="$t('settings.linksCategoriesLimit')"
          )
          v-divider
          export-csv-form(v-model="settings.widget.parameters", datetime-format)
          v-divider
          field-switcher(
            v-model="settings.widget.parameters.sticky_header",
            :title="$t('settings.stickyHeader')"
          )
      v-divider
    v-btn.primary(data-test="submitAlarms", @click="submit") {{ $t('common.save') }}
</template>

<script>
import { get } from 'lodash';

import { PAGINATION_LIMIT } from '@/config';
import { SIDE_BARS } from '@/constants';

import { alarmListWidgetToForm, formToAlarmListWidget } from '@/helpers/forms/widgets/alarm';

import { widgetSettingsMixin } from '@/mixins/widget/settings';
import { permissionsWidgetsAlarmsListFilters } from '@/mixins/permissions/widgets/alarms-list/filters';
import { permissionsWidgetsAlarmsListRemediationInstructionsFilters }
  from '@/mixins/permissions/widgets/alarms-list/remediation-instructions-filters';

import FieldTitle from './fields/common/title.vue';
import FieldDefaultSortColumn from './fields/common/default-sort-column.vue';
import FieldColumns from './fields/common/columns.vue';
import FieldLiveReporting from './fields/common/live-reporting.vue';
import FieldPeriodicRefresh from './fields/common/periodic-refresh.vue';
import FieldDefaultElementsPerPage from './fields/common/default-elements-per-page.vue';
import FieldFilters from './fields/common/filters.vue';
import FieldTextEditor from './fields/common/text-editor.vue';
import FieldSwitcher from './fields/common/switcher.vue';
import FieldFastAckOutput from './fields/alarm/fast-ack-output.vue';
import FieldGridRangeSize from './fields/common/grid-range-size.vue';
import FieldRemediationInstructionsFilters from './fields/common/remediation-instructions-filters.vue';
import FieldOpenedResolvedFilter from './fields/alarm/opened-resolved-filter.vue';
import FieldInfoPopup from './fields/alarm/info-popup.vue';
import FieldEnabledLimit from './fields/common/enabled-limit.vue';
import ExportCsvForm from './forms/export-csv.vue';

/**
 * Component to regroup the alarms list settings fields
 */
export default {
  name: SIDE_BARS.alarmSettings,
  $_veeValidate: {
    validator: 'new',
  },
  components: {
    FieldTitle,
    FieldDefaultSortColumn,
    FieldColumns,
    FieldLiveReporting,
    FieldPeriodicRefresh,
    FieldDefaultElementsPerPage,
    FieldOpenedResolvedFilter,
    FieldFilters,
    FieldTextEditor,
    FieldSwitcher,
    FieldFastAckOutput,
    FieldGridRangeSize,
    FieldRemediationInstructionsFilters,
    FieldInfoPopup,
    FieldEnabledLimit,
    ExportCsvForm,
  },
  mixins: [
    widgetSettingsMixin,
    permissionsWidgetsAlarmsListFilters,
    permissionsWidgetsAlarmsListRemediationInstructionsFilters,
  ],
  data() {
    const { widget } = this.config;

    return {
      settings: {
        widget: alarmListWidgetToForm(widget),
        userPreferenceContent: { itemsPerPage: PAGINATION_LIMIT },
      },
    };
  },
  mounted() {
    const { content } = this.userPreference;

    this.settings.userPreferenceContent = {
      itemsPerPage: get(content, 'itemsPerPage', PAGINATION_LIMIT),
    };
  },
  methods: {
    updateMainFilterUpdatedAt() {
      this.settings.widget.parameters.mainFilterUpdatedAt = Date.now();
    },

    prepareWidgetSettings() {
      const { widget } = this.settings;

      return formToAlarmListWidget(widget);
    },
  },
};
</script>
