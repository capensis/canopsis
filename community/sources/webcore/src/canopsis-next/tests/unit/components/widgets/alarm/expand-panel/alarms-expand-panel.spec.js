import { generateShallowRenderer, generateRenderer } from '@unit/utils/vue';

import { createMockedStoreModule, createMockedStoreModules } from '@unit/utils/store';

import { fakeAlarmDetails } from '@unit/data/alarm';

import {
  CANOPSIS_EDITION,
  ENTITY_TYPES,
  JUNIT_ALARM_CONNECTOR,
  USERS_PERMISSIONS,
} from '@/constants';

import AlarmsExpandPanel from '@/components/widgets/alarm/expand-panel/alarms-expand-panel.vue';

const stubs = {
  'more-infos': true,
  'alarms-time-line': true,
  'alarms-expand-panel-children': true,
  'service-dependencies': true,
  'declared-tickets-list': true,
  'entity-gantt': true,
  'pbehaviors-simple-list': true,
};

const selectTabs = wrapper => wrapper.vm.$children[0];

describe('alarms-expand-panel', () => { // TODO: add tests for children, timeline, query
  const infoModule = {
    name: 'info',
    getters: { edition: CANOPSIS_EDITION.community },
  };

  const catInfoModule = {
    name: 'info',
    getters: { edition: CANOPSIS_EDITION.pro },
  };
  const authModule = {
    name: 'auth',
    getters: {
      currentUserPermissionsById: {
        [USERS_PERMISSIONS.technical.exploitation.pbehavior]: { actions: [] },
      },
    },
  };

  const fetchAlarmDetails = jest.fn();
  const fetchAlarmsDetailsList = jest.fn();
  const updateAlarmDetailsQuery = jest.fn();
  const removeAlarmDetailsQuery = jest.fn();

  const alarmDetailsModule = createMockedStoreModule({
    name: 'details',
    getters: {
      getItem: () => () => fakeAlarmDetails(),
      getPending: () => () => false,
      getQuery: () => () => ({ page: 1, limit: 10 }),
      getQueries: () => () => [
        { page: 2, limit: 5 },
        { page: 1, limit: 10 },
      ],
    },
    actions: {
      fetchItem: fetchAlarmDetails,
      fetchList: fetchAlarmsDetailsList,
      updateQuery: updateAlarmDetailsQuery,
      removeQuery: removeAlarmDetailsQuery,
    },
  });

  const alarmModule = {
    name: 'alarm',
    modules: {
      details: alarmDetailsModule,
    },
  };

  const store = createMockedStoreModules([
    authModule,
    alarmModule,
    infoModule,
  ]);

  const factory = generateShallowRenderer(AlarmsExpandPanel, {
    stubs,
  });
  const snapshotFactory = generateRenderer(AlarmsExpandPanel, {
    stubs,
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Tabs key updated after change tour enabled', async () => {
    const wrapper = factory({
      store,
      propsData: {
        alarm: {
          _id: 'alarm-id',
          entity: {
            type: ENTITY_TYPES.connector,
            impact: [],
          },
          v: {},
        },
        widget: {
          parameters: {
            moreInfoTemplate: 'template',
            isHtmlEnabledOnTimeLine: false,
            serviceDependenciesColumns: [],
          },
        },
      },
    });

    const prevKey = selectTabs(wrapper).$vnode.key;

    await wrapper.setProps({
      isTourEnabled: true,
    });

    expect(prevKey !== selectTabs(wrapper).$vnode.key).toBe(true);
  });

  it('Tab key updated after change moreInfoTemplate', async () => {
    const widget = {
      parameters: {
        moreInfoTemplate: 'template',
      },
    };
    const wrapper = factory({
      store,
      propsData: {
        alarm: {
          _id: 'alarm-id',
          entity: {
            type: ENTITY_TYPES.connector,
            impact: [],
          },
          v: {},
        },
        widget,
      },
    });

    const prevKey = selectTabs(wrapper).$vnode.key;

    await wrapper.setProps({
      widget: {
        parameters: {
          moreInfoTemplate: 'template2',
        },
      },
    });

    expect(prevKey !== selectTabs(wrapper).$vnode.key).toBe(true);
  });

  it('Renders `alarms-expand-panel` with required props', () => {
    const wrapper = snapshotFactory({
      store,
      propsData: {
        alarm: {
          _id: 'alarm-id',
          entity: {
            type: ENTITY_TYPES.connector,
            impact: [],
          },
          v: {},
        },
        widget: {
          parameters: {
            moreInfoTemplate: 'template',
            isHtmlEnabledOnTimeLine: false,
            serviceDependenciesColumns: [],
          },
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `alarms-expand-panel` with custom props', () => {
    const wrapper = snapshotFactory({
      store,
      propsData: {
        alarm: {
          _id: 'alarm-id',
          causes: {},
          consequences: {},
          entity: {
            type: ENTITY_TYPES.connector,
            impact: [],
          },
          v: {},
        },
        widget: {
          parameters: {
            moreInfoTemplate: 'template',
            isHtmlEnabledOnTimeLine: false,
            serviceDependenciesColumns: [],
          },
        },
        hideChildren: true,
        editing: true,
        isTourEnabled: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `alarms-expand-panel` with full alarm', () => {
    const wrapper = snapshotFactory({
      store,
      propsData: {
        alarm: {
          _id: 'alarm-id',
          causes: {},
          consequences: {},
          entity: {
            type: ENTITY_TYPES.service,
            impact: ['test'],
          },
          v: {
            tickets: [{}],
          },
        },
        widget: {
          parameters: {
            moreInfoTemplate: 'template',
            isHtmlEnabledOnTimeLine: false,
            serviceDependenciesColumns: [{}],
          },
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `alarms-expand-panel` with gantt', () => {
    const wrapper = snapshotFactory({
      store: createMockedStoreModules([
        alarmModule,
        catInfoModule,
        authModule,
      ]),
      propsData: {
        alarm: {
          _id: 'alarm-id',
          v: {
            connector: JUNIT_ALARM_CONNECTOR,
          },
          entity: {
            type: ENTITY_TYPES.resource,
            impact: [],
          },
        },
        widget: {
          parameters: {
            moreInfoTemplate: 'template',
            isHtmlEnabledOnTimeLine: false,
            serviceDependenciesColumns: [{}],
          },
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
