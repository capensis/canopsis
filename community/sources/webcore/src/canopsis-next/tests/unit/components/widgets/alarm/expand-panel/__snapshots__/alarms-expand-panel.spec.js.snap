// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alarms-expand-panel Renders \`alarms-expand-panel\` with custom props 1`] = `
<div
  class="v-tabs expand-panel secondary lighten-2 v-tabs--centered theme--dark"
  editing="true"
>
  <div
    class="v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text secondary lighten-1"
    role="tablist"
  >
    <div
      class="v-slide-group__prev v-slide-group__prev--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
    <div
      class="v-slide-group__wrapper"
    >
      <div
        class="v-slide-group__content v-tabs-bar__content"
      >
        <div
          class="v-tabs-slider-wrapper"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="v-tabs-slider primary"
          />
        </div>
        <a
          aria-selected="true"
          class="v-tab v-tab--active"
          href="#moreInfos"
          role="tab"
          tabindex="0"
        >
    More infos
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#timeLine"
          role="tab"
          tabindex="0"
        >
    Timeline
        </a>
        <!---->
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#pbehavior"
          role="tab"
          tabindex="0"
        >
    Pbehaviors
        </a>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div
      class="v-slide-group__next v-slide-group__next--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
  </div>
  <div
    class="v-window v-item-group theme--light v-tabs-items"
  >
    <div
      class="v-window__container"
    >
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="moreInfos"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <div
                    class="more-infos"
                  >
                    <div>
                      template
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="timeLine"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-progress-linear--visible theme--light"
                  role="progressbar"
                  style="height: 0px;"
                >
                  <div
                    class="v-progress-linear__background primary"
                    style="opacity: 0.3; left: 0%; width: 100%;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="height: 0px;"
                  />
                  <transition-stub
                    name="fade-transition"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      <div
                        class="v-progress-linear__indeterminate long primary"
                      />
                      <div
                        class="v-progress-linear__indeterminate short primary"
                      />
                    </div>
                  </transition-stub>
                </div>
                <div
                  class="v-card__text"
                >
                  <alarms-time-line-stub
                    steps="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="pbehavior"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <pbehaviors-simple-list-stub
                    entity="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`alarms-expand-panel Renders \`alarms-expand-panel\` with full alarm 1`] = `
<div
  class="v-tabs expand-panel secondary lighten-2 v-tabs--centered theme--dark"
>
  <div
    class="v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text secondary lighten-1"
    role="tablist"
  >
    <div
      class="v-slide-group__prev v-slide-group__prev--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
    <div
      class="v-slide-group__wrapper"
    >
      <div
        class="v-slide-group__content v-tabs-bar__content"
      >
        <div
          class="v-tabs-slider-wrapper"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="v-tabs-slider primary"
          />
        </div>
        <a
          aria-selected="true"
          class="v-tab v-tab--active"
          href="#moreInfos"
          role="tab"
          tabindex="0"
        >
    More infos
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#timeLine"
          role="tab"
          tabindex="0"
        >
    Timeline
        </a>
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#ticketsDeclared"
          role="tab"
          tabindex="0"
        >
    Tickets declared
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#pbehavior"
          role="tab"
          tabindex="0"
        >
    Pbehaviors
        </a>
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#trackSource"
          role="tab"
          tabindex="0"
        >
    Track source
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#impactChain"
          role="tab"
          tabindex="0"
        >
    Impact chain
        </a>
        <!---->
      </div>
    </div>
    <div
      class="v-slide-group__next v-slide-group__next--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
  </div>
  <div
    class="v-window v-item-group theme--light v-tabs-items"
  >
    <div
      class="v-window__container"
    >
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="moreInfos"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <div
                    class="more-infos"
                  >
                    <div>
                      template
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="timeLine"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-progress-linear--visible theme--light"
                  role="progressbar"
                  style="height: 0px;"
                >
                  <div
                    class="v-progress-linear__background primary"
                    style="opacity: 0.3; left: 0%; width: 100%;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="height: 0px;"
                  />
                  <transition-stub
                    name="fade-transition"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      <div
                        class="v-progress-linear__indeterminate long primary"
                      />
                      <div
                        class="v-progress-linear__indeterminate short primary"
                      />
                    </div>
                  </transition-stub>
                </div>
                <div
                  class="v-card__text"
                >
                  <alarms-time-line-stub
                    steps="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="ticketsDeclared"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <declared-tickets-list-stub
                    tickets="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="pbehavior"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <pbehaviors-simple-list-stub
                    entity="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="trackSource"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <service-dependencies-stub
                    columns="[object Object]"
                    includeroot="true"
                    openableroot="true"
                    root="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="impactChain"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <service-dependencies-stub
                    columns="[object Object]"
                    impact="true"
                    includeroot="true"
                    openableroot="true"
                    root="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
</div>
`;

exports[`alarms-expand-panel Renders \`alarms-expand-panel\` with gantt 1`] = `
<div
  class="v-tabs expand-panel secondary lighten-2 v-tabs--centered theme--dark"
>
  <div
    class="v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text secondary lighten-1"
    role="tablist"
  >
    <div
      class="v-slide-group__prev v-slide-group__prev--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
    <div
      class="v-slide-group__wrapper"
    >
      <div
        class="v-slide-group__content v-tabs-bar__content"
      >
        <div
          class="v-tabs-slider-wrapper"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="v-tabs-slider primary"
          />
        </div>
        <a
          aria-selected="true"
          class="v-tab v-tab--active"
          href="#moreInfos"
          role="tab"
          tabindex="0"
        >
    More infos
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#timeLine"
          role="tab"
          tabindex="0"
        >
    Timeline
        </a>
        <!---->
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#pbehavior"
          role="tab"
          tabindex="0"
        >
    Pbehaviors
        </a>
        <!---->
        <!---->
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#entityGantt"
          role="tab"
          tabindex="0"
        >
    Gantt chart
        </a>
      </div>
    </div>
    <div
      class="v-slide-group__next v-slide-group__next--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
  </div>
  <div
    class="v-window v-item-group theme--light v-tabs-items"
  >
    <div
      class="v-window__container"
    >
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="moreInfos"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <div
                    class="more-infos"
                  >
                    <div>
                      template
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="timeLine"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-progress-linear--visible theme--light"
                  role="progressbar"
                  style="height: 0px;"
                >
                  <div
                    class="v-progress-linear__background primary"
                    style="opacity: 0.3; left: 0%; width: 100%;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="height: 0px;"
                  />
                  <transition-stub
                    name="fade-transition"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      <div
                        class="v-progress-linear__indeterminate long primary"
                      />
                      <div
                        class="v-progress-linear__indeterminate short primary"
                      />
                    </div>
                  </transition-stub>
                </div>
                <div
                  class="v-card__text"
                >
                  <alarms-time-line-stub
                    steps="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="pbehavior"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <pbehaviors-simple-list-stub
                    entity="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="entityGantt"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <entity-gantt-stub
                    alarm="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`alarms-expand-panel Renders \`alarms-expand-panel\` with required props 1`] = `
<div
  class="v-tabs expand-panel secondary lighten-2 v-tabs--centered theme--dark"
>
  <div
    class="v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text secondary lighten-1"
    role="tablist"
  >
    <div
      class="v-slide-group__prev v-slide-group__prev--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
    <div
      class="v-slide-group__wrapper"
    >
      <div
        class="v-slide-group__content v-tabs-bar__content"
      >
        <div
          class="v-tabs-slider-wrapper"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="v-tabs-slider primary"
          />
        </div>
        <a
          aria-selected="true"
          class="v-tab v-tab--active"
          href="#moreInfos"
          role="tab"
          tabindex="0"
        >
    More infos
        </a>
        <a
          aria-selected="false"
          class="v-tab"
          href="#timeLine"
          role="tab"
          tabindex="0"
        >
    Timeline
        </a>
        <!---->
        <!---->
        <a
          aria-selected="false"
          class="v-tab"
          href="#pbehavior"
          role="tab"
          tabindex="0"
        >
    Pbehaviors
        </a>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div
      class="v-slide-group__next v-slide-group__next--disabled"
    >
      <transition-stub
        name="fade-transition"
      />
    </div>
  </div>
  <div
    class="v-window v-item-group theme--light v-tabs-items"
  >
    <div
      class="v-window__container"
    >
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="moreInfos"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <div
                    class="more-infos"
                  >
                    <div>
                      template
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="timeLine"
        >
          <div
            class="layout pa-3"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-progress-linear--visible theme--light"
                  role="progressbar"
                  style="height: 0px;"
                >
                  <div
                    class="v-progress-linear__background primary"
                    style="opacity: 0.3; left: 0%; width: 100%;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="height: 0px;"
                  />
                  <transition-stub
                    name="fade-transition"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      <div
                        class="v-progress-linear__indeterminate long primary"
                      />
                      <div
                        class="v-progress-linear__indeterminate short primary"
                      />
                    </div>
                  </transition-stub>
                </div>
                <div
                  class="v-card__text"
                >
                  <alarms-time-line-stub
                    steps="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <transition-stub
        name="v-window-x-transition"
      >
        <div
          class="v-window-item v-window-item--active"
          id="pbehavior"
        >
          <div
            class="layout pa-3 secondary lighten-2"
          >
            <div
              class="flex offset-xs0 xs12"
            >
              <div
                class="tab-item-card v-card v-sheet theme--light"
              >
                <div
                  class="v-card__text"
                >
                  <pbehaviors-simple-list-stub
                    entity="[object Object]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
